src/all/taddyink/src/eu/kanade/tachiyomi/extension/all/taddyink/TaddyUtils.kt

```diff
@@ -0,0 +1,144 @@
+package eu.kanade.tachiyomi.extension.all.taddyink
+
+import eu.kanade.tachiyomi.source.model.SManga
+import kotlinx.serialization.Serializable
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+object TaddyUtils {
+    private val formatter = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.US)
+
+    fun getManga(comicObj: Comic): SManga {
+        val name = comicObj.name
+        val sssUrl = comicObj.url
+        val sssDescription = comicObj.description
+        val genres = comicObj.genres?.map { genre ->
+            genreMap[genre] ?: ""
+        }?.joinToString(", ")
+
+        val creators = comicObj.creators?.mapNotNull { creator ->
+            creator.name
+        }?.joinToString(", ")
+
+        val thumbnailBaseUrl = comicObj.coverImage?.base_url ?: ""
+        val thumbnail = comicObj.coverImage?.cover_sm ?: ""
+        val thumbnailUrl = if (thumbnailBaseUrl.isNotEmpty() && thumbnail.isNotEmpty()) "$thumbnailBaseUrl$thumbnail" else ""
+
+        return SManga.create().apply {
+            url = sssUrl
+            title = name
+            creators?.takeIf { it.isNotBlank() }?.let { author = it }
```

Comment:
I don't remember at the moment if `author` and `artist` accepts `null`. If it accepts:

```suggestion
            author = creators
```

If not:

```suggestion
            author = creators.orEmpty()
```

Reply:
It will be set as Unknown if the user manually refreshes the details, so this isn't much of an issue.

Reply:
I tested refreshing, and it keeps the old value. Maybe I missed a case on how to refresh.

But right now the main screen gets the creator name, you go to the comic (and you see the creator name), you pull to reload (and the creator name stays)